sdks/go/test/integration/wordcount/wordcount_test.go

```diff
@@ -75,19 +87,26 @@ func TestWordCount(t *testing.T) {
 			},
 			2,
 			"Nz70m/sn3Ep9o484r7MalQ==", // whitespace doesn't matter: same hash as above
+			"smallWords",
+			1,
 		},
 	}
 
 	for _, test := range tests {
 		integration.CheckFilters(t)
-		const filename = "memfs://input"
-		memfs.Write(filename, []byte(strings.Join(test.lines, "\n")))
-
-		p := WordCount(filename, test.hash, test.words)
-		_, err := ptest.RunWithMetrics(p)
+		p, s := beam.NewPipelineWithRoot()
+		lines := beam.CreateList(s, test.lines)
+		WordCountFromPCol(s, lines, test.hash, test.words)
+		pr, err := ptest.RunWithMetrics(p)
 		if err != nil {
 			t.Errorf("WordCount(\"%v\") failed: %v", strings.Join(test.lines, "|"), err)
 		}
+		qr := pr.Metrics().Query(func(sr metrics.SingleResult) bool {
+			return sr.Name() == test.name
+		})
+		if len(qr.Counters()) != test.counters {
+			t.Errorf("Metrics filtering with Name failed.\nGot %d counters \n Want %d counters", len(qr.Counters()), test.counters)
```

Comment:
As a rule, if you don't have to, don't spread test output out like this. This case is going to change from the descriptions above but the following reads better and says what method failed.
```
"Metrics().Query(by Name) failed. Got %d counters, want %d counters"
```